src/icon-loader.c

```diff
@@ -82,6 +83,60 @@ icon_loader_finish(struct server *server)
 	server->icon_loader = NULL;
 }
 
+struct icon_ctx {
+	char *path;
+	enum sfdo_icon_file_format format;
+};
+
+/* Return 0 on success and -1 on error */
+static int
+process_rel_name(struct icon_ctx *ctx, const char *icon_name,
+		struct icon_loader *loader, int size, int scale)
+{
+	int ret = 0;
+	int lookup_options = SFDO_ICON_THEME_LOOKUP_OPTIONS_DEFAULT;
+#if !HAVE_RSVG
+	lookup_options |= SFDO_ICON_THEME_LOOKUP_OPTION_NO_SVG;
+#endif
+	struct sfdo_icon_file *icon_file = sfdo_icon_theme_lookup(
+		loader->icon_theme, icon_name, SFDO_NT, size, scale,
+		lookup_options);
+	if (!icon_file || icon_file == SFDO_ICON_FILE_INVALID) {
+		ret = -1;
+		goto out;
+	}
+	const char *path = sfdo_icon_file_get_path(icon_file, NULL);
+	if (string_null_or_empty(path)) {
```

Comment:
This should always be `false`.

Reply:
Okay, have just gone for a straight `xstrdup(sfdo_icon_file_get_path())`

I did actually look at the `sfdo_icon_file_get_path()` source - but I'm just paranoid :smile: I'll try to relax.